ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+
+        return None
+
+    def _get_table_fqn_from_om(self, table_details: TableDetails) -> Optional[str]:
+        """
+        Based on partial schema and table names look for matching table object in open metadata.
+        :param schema: schema name
+        :param table: table name
+        :return: fully qualified name of a Table in Open Metadata
+        """
+        result = None
+        services = self.source_config.dbServiceNames
+        for db_service in services:
+            result = self._get_entity_fqn_from_om(
+                Table,
+                service_name=db_service,
+                table_name=table_details.name,
+                schema_name=table_details.schema,
+                database_name=None,
+            )
+            if result:
+                return result
+        if not result:
+            raise FQNNotFoundException(
+                f"Table FQN not found for table: {table_details} within services: {services}"
+            )
+        return result
+
+    def _get_schema_fqn_from_om(self, schema: str) -> Optional[str]:
+        """
+        Based on partial schema name look for any matching DatabaseSchema object in open metadata.
+
+        :param schema: schema name
+        :return: fully qualified name of a DatabaseSchema in Open Metadata
+        """
+        result = None
+        services = self.source_config.dbServiceNames
+
+        for db_service in services:
+            databases = [
+                x.get("name")
+                for x in self.metadata.client.get(
+                    "/databases",
+                    data={
+                        "service": db_service,
+                        "include": "non-deleted",
+                        "limit": 1000000,
+                    },
+                ).get("data", [])
+            ]
+
+            for database in databases:
+                result = self._get_entity_fqn_from_om(
+                    DatabaseSchema,
+                    service_name=db_service,
+                    schema_name=schema,
+                    database_name=database,
+                )
+
+                if result:
+                    return result
+
+        if not result:
+            raise FQNNotFoundException(
+                f"Schema FQN not found within services: {services}"
+            )
+
+        return result
+
+    @classmethod
+    def _render_pipeline_name(cls, pipeline_details: OpenLineageEvent) -> str:
+        """
+        Renders pipeline name from parent facet of run facet. It is our expectation that every OL event contains parent
+        run facet so we can always create pipeline entities and link them to lineage events.
+
+        :param run_facet: Open Lineage run facet
+        :return: pipeline name (not fully qualified name)
+        """
+        run_facet = pipeline_details.run_facet
+
+        namespace = run_facet["facets"]["parent"]["job"]["namespace"]
+        name = run_facet["facets"]["parent"]["job"]["name"]
```

Comment:
in the OpenLineageEvent we have defined `run_facet` of  generic `Dict` type do you think we can be more specific and define the exact model also to avoid keyerror in case when we did not receive  the response in expected format.

Reply:
+1 on this, will make maintaining and evolving things easier

Reply:
bump